### ISSUE TITLE
Extension of the Order Attribute

### ISSUE THREAD

**Issue by NUnitTester** [nunit/nunit#2108]
I like the Order-Attribute very much. But I want to differ between two cases of ordered tests: 
- I like a order because it is nice to have one
- I need a order because all tests with higher value will fail.

For the first case it is enough. For the second case I need another behaviour. I want to write [Order(1, Behaviour.Critical)] that means, if this test fails all tests with an higher Order (or no Order-Attribute) will not run at all and are inconclusive. The other thing if nothing is written is Behaviour.Normal.

example:
public class IntegrationTest
{
  [Test, Order(1, Behaviour.Critical)]
  public void TestDatabaseAvailable()
  { ... }   // if this test fails the database is not available

  [Test, Order(2, Behaviour.Critical)]
  public void TestUserConnect()
  { ... }    // if this test fails the user cannot login

  [Test]
  public void TestSomthing1()
  { ... }   // normal Tests

  [Test]
  public void TestSomthing2()
  { ... }   // normal Tests
}

**Comment by CharliePoole** [comment]
@nunit/core-team @nunit/framework-team 

I'm not happy with supporting the writing of dependent tests this way, unless we are going to define some sort of "IntegrationFixture" as a major feature.

OTOH, I could live with a separate attribute like "CriticalFailure" that caused the fixture to be abandoned if the test failed. I could also live with not doing this at all. 😄

**Comment by ChrisMaddock** [comment]
This sounds more fitting to be an addition to a [DependencyAttribute](https://github.com/nunit/nunit/issues/51) - rather than related to OrderAttribute.

You're saying that test 2 is dependent on test 1, and if that fails, then abort all tests. That is the behaviour I'd expect if a 'dependent test' failed, with the hypothetical DependencyAttribute.

**Comment by rprouse** [comment]
I also think this would be better served by a DependecyAttribute and think this is outside the scope and purpose of the OrderAttribute. I would vote to close this.

**Comment by NUnitTester** [comment]
Not sure: See the Entry of  circa1741 commented on 28 Nov 2014 of the Feature "DependencyAttribute".
I think it is better to do with the Order because the Dependency needs a method name and it is easier to use order. There could be a amount of methods for one level. And only if one of a lower level failed (and in my view is also critical) than the levels above should not be executed.
It is to many work to give a list of methods names, I think.

And in [OrderAttribute](https://github.com/nunit/nunit/issues/170) you wrote
( CharliePoole commented on 8 Aug 2014 )
- Don't start test B until after you start test A
- Don't start test B until after test A completes
- **Don't start test B until test A completes successfully**  AND that is missing.

But I dont think it is always good to do so. It depends on the special situation, in some cases ist is needed and in some not. So I want to have this configurable. And one solution would be that one I have wrote above. But a new Attribute i soo much, I think
